### PR TITLE
[Old Draft] 8260931: Implement JEP 382: New macOS Rendering Pipeline

### PR DESCRIPTION
**Pull request by aghaisas**
This is the implementation of JEP 382 : New macOS Rendering Pipeline (Metal rendering pipeline)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [ ] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Contributors
 * Ajit Ghaisas `<aghaisas@openjdk.org>`

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2361/head:pull/2361`
`$ git checkout pull/2361`

### CODE REVIEW

File: src/java.desktop/macosx/classes/sun/java2d/metal/MTLBlitLoops.java [hunk 1]
```diff
@@ -0,0 +1,959 @@
...
+        buf.putInt(packedParams);
+        buf.putInt(sx1).putInt(sy1);
+        buf.putInt(sx2).putInt(sy2);
+        buf.putDouble(dx1).putDouble(dy1);
+        buf.putDouble(dx2).putDouble(dy2);
+        buf.putLong(src.getNativeOps());
+        buf.putLong(dst.getNativeOps());
+    }
+
+    static void Blit(SurfaceData srcData, SurfaceData dstData,
```
Comment (aimiebell) [inline src/java.desktop/macosx/classes/sun/java2d/metal/MTLBlitLoops.java:262]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: src/java.desktop/macosx/classes/sun/java2d/metal/MTLBufImgOps.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        MTLGraphicsConfig gc = mtlSrc.getMTLGraphicsConfig();
+        if (mtlSrc.getType() != MTLSurfaceData.TEXTURE ||
+                !gc.isCapPresent(CAPS_EXT_BIOP_SHADER))
+        {
+            return false;
+        }
+
+        int sw = img.getWidth();
+        int sh = img.getHeight();
+        MTLBlitLoops.IsoBlit(srcData, dstData,
```
Comment (aimiebell) [inline src/java.desktop/macosx/classes/sun/java2d/metal/MTLBufImgOps.java:107]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: src/java.desktop/macosx/classes/sun/java2d/metal/MTLDrawImage.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                            interpType == AffineTransformOp.TYPE_NEAREST_NEIGHBOR))
+            {
+                SurfaceType srcType = srcData.getSurfaceType();
+                SurfaceType dstType = dstData.getSurfaceType();
+                TransformBlit blit = TransformBlit.getFromCache(srcType,
+                        sg.imageComp,
+                        dstType);
+
+                if (blit != null) {
+                    blit.Transform(srcData, dstData,
```
Comment (aimiebell) [inline src/java.desktop/macosx/classes/sun/java2d/metal/MTLDrawImage.java:81]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Comment by jayathirthrao** [comment]
@Ajit i think we can close this draft PR. It is coming up in pending PR's of 2d.

**Pull request closed without merging**